File: examples/java/src/c/ConjAsFunction.java
package c;

public class ConjAsFunction {
  public static final long conjMapMap() {
    final Object a =
        new io.lacuna.bifurcan.Map<Object, Object>()
            .put(":a", 1L, io.lacuna.bifurcan.Maps.MERGE_LAST_WRITE_WINS);
    final Object b =
        new io.lacuna.bifurcan.Map<Object, Object>()
            .put(":b", 2L, io.lacuna.bifurcan.Maps.MERGE_LAST_WRITE_WINS);
    final Object c = kalai.Kalai.conj(a, b);
    return 3L;
  }

  public static final long conjMapVec() {
    final Object a =
        new io.lacuna.bifurcan.Map<Object, Object>()
            .put(":a", 1L, io.lacuna.bifurcan.Maps.MERGE_LAST_WRITE_WINS);
    final Object b = new io.lacuna.bifurcan.List<Object>().addLast(":b").addLast(2L);
    final Object c = kalai.Kalai.conj(a, b);
    return 11L;
  }

  public static final long conjSet() {
    final Object a = new io.lacuna.bifurcan.Set<Object>().add(":a").add(":b");
    final Object newValue = ":c";
    final Object c = kalai.Kalai.conj(a, newValue);
    return 5L;
  }

  public static final long conjVector() {
    final Object a = new io.lacuna.bifurcan.List<Object>().addLast(":a").addLast(":b");
    final Object newValue = ":c";
    final Object c = kalai.Kalai.conj(a, newValue);
    return 7L;
  }

  public static final long typeConversions() {
    final io.lacuna.bifurcan.Map<Object, Object> a =
        new io.lacuna.bifurcan.Map<Object, Object>()
            .put(":a", 1L, io.lacuna.bifurcan.Maps.MERGE_LAST_WRITE_WINS);
    final io.lacuna.bifurcan.Map<Object, Object> b =
        new io.lacuna.bifurcan.Map<Object, Object>()
            .put(":b", 1L, io.lacuna.bifurcan.Maps.MERGE_LAST_WRITE_WINS);
    {
      kalai.Kalai.conj((Object) a, (Object) b);
      return 4L;
    }
  }

  public static final void main(String[] _args) {
    System.out.println(conjMapMap());
    System.out.println(conjMapVec());
    System.out.println(conjSet());
    System.out.println(conjVector());
    System.out.println(typeConversions());
  }
}

File: examples/rust/src/c/conj_as_function.rs
use crate::kalai::kalai;
use crate::kalai::kalai::*;
pub fn conj_map_map() -> i64 {
    let a: kalai::BValue = kalai::BValue::from(rpds::HashTrieMap::new().insert(
        kalai::BValue::from(String::from(":a")),
        kalai::BValue::from(1i64),
    ));
    let b: kalai::BValue = kalai::BValue::from(rpds::HashTrieMap::new().insert(
        kalai::BValue::from(String::from(":b")),
        kalai::BValue::from(2i64),
    ));
    let c: kalai::BValue = conj(a, b);
    return 3i64;
}
pub fn conj_map_vec() -> i64 {
    let a: kalai::BValue = kalai::BValue::from(rpds::HashTrieMap::new().insert(
        kalai::BValue::from(String::from(":a")),
        kalai::BValue::from(1i64),
    ));
    let b: kalai::BValue = kalai::BValue::from(
        rpds::Vector::new()
            .push_back(kalai::BValue::from(String::from(":b")))
            .push_back(kalai::BValue::from(2i64)),
    );
    let c: kalai::BValue = conj(a, b);
    return 11i64;
}
pub fn conj_set() -> i64 {
    let a: kalai::BValue = kalai::BValue::from(
        rpds::HashTrieSet::new()
            .insert(kalai::BValue::from(String::from(":a")))
            .insert(kalai::BValue::from(String::from(":b"))),
    );
    let new_value: kalai::BValue = kalai::BValue::from(String::from(":c"));
    let c: kalai::BValue = conj(a, new_value);
    return 5i64;
}
pub fn conj_vector() -> i64 {
    let a: kalai::BValue = kalai::BValue::from(
        rpds::Vector::new()
            .push_back(kalai::BValue::from(String::from(":a")))
            .push_back(kalai::BValue::from(String::from(":b"))),
    );
    let new_value: kalai::BValue = kalai::BValue::from(String::from(":c"));
    let c: kalai::BValue = conj(a, new_value);
    return 7i64;
}
pub fn type_conversions() -> i64 {
    let a: rpds::HashTrieMap<kalai::BValue, kalai::BValue> = rpds::HashTrieMap::new().insert(
        kalai::BValue::from(String::from(":a")),
        kalai::BValue::from(1i64),
    );
    let b: rpds::HashTrieMap<kalai::BValue, kalai::BValue> = rpds::HashTrieMap::new().insert(
        kalai::BValue::from(String::from(":b")),
        kalai::BValue::from(1i64),
    );
    {
        conj(kalai::BValue::from(a), kalai::BValue::from(b));
        return 4i64;
    }
}
pub fn main() {
    let _args: std::vec::Vec<String> = std::env::args().collect();
    {
        println!("{}", conj_map_map());
        println!("{}", conj_map_vec());
        println!("{}", conj_set());
        println!("{}", conj_vector());
        println!("{}", type_conversions());
    }
}
